grid-proxy/internal/indexer/gpu.go

```diff
@@ -37,16 +31,15 @@ type NodeGPUIndexer struct {
 
 func NewNodeGPUIndexer(
 	ctx context.Context,
-	relayURL,
-	mnemonics string,
-	subManager substrate.Manager,
+	rpcClient *peer.RpcClient,
 	db db.Database,
 	indexerCheckIntervalMins,
 	batchSize,
 	nodesGPUResultsWorkers,
-	nodesGPUBufferWorkers uint) (*NodeGPUIndexer, error) {
-	indexer := &NodeGPUIndexer{
+	nodesGPUBufferWorkers uint) *NodeGPUIndexer {
+	return &NodeGPUIndexer{
 		db:                     db,
+		rpcClient:              rpcClient,
```

Comment:
is that client a reconnecting one, in long running processes you should be careful not to lose that single connection you have.

Reply:
afaik yes, the rpcClient taking a substrate manager and handle the open/close connections when needed